### PR TITLE
feat: [OeX_Cred-164] added 'IssuanceConfiguration'

### PR DESCRIPTION
**Pull request by Thposbsmt**
**Run JavaScript tests locally with Karma**

There is work being done on a fix to get Karma to run in CI. Until then, however, contributors are required to run these tests locally.

- [x] Make sure you are inside the devstack container
- [x] Run `make test-karma`
- [x] All tests pass

### CODE REVIEW

**Comment by Thposbsmt** [comment]
It need to have following dependencies to create `IssuanceConfiguration`, such as organizations and config model. Requirements file requires to run `make upgrade` after adding new dependencies

File: credentials/apps/verifiable_credentials/models.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        default="default",
+        blank=True,
+    )
+    organization = models.ForeignKey(
+        Organization,
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+    )
+    digital_credential_type = models.CharField(
```
Comment (wowkalucky) [inline credentials/apps/verifiable_credentials/models.py:48]: We decided to use `format` over `type` everywhere.

File: credentials/apps/verifiable_credentials/models.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        max_length=128,
+        blank=True,
+        verbose_name="Digital Credential Type",
+        default=VERIFIABLE_CREDENTIAL_KEY,
+        choices=DIGITAL_CREDENTIAL_TYPE_CHOICES,
```
Comment (wowkalucky) [inline credentials/apps/verifiable_credentials/models.py:53]: ditto

File: credentials/settings/base.py [hunk 3]
```diff
@@ -556,3 +559,16 @@
...
+# .. toggle_name: ENABLE_VERIFIABLE_CREDENTIALS
+# .. toggle_implementation: DjangoSetting
+# .. toggle_default: False
+# .. toggle_description: Determines if the Credentials IDA uses digital credentials issuance.
+# .. toggle_warning: Requires the Learner Record MFE to be deployed and used in a given environment if toggled to true.
+# .. toggle_life_expectancy: permanent
+# .. toggle_permanent_justification: Digital Credentials are optional for usage.
+# .. toggle_creation_date: 2023-02-02
+ENABLE_VERIFIABLE_CREDENTIALS = False
+VC_DEFAULT_TYPE = "VC"
```
Comment (wowkalucky) [inline credentials/settings/base.py:574]: This should go into [inner settings](https://youtrack.raccoongang.com/issue/OeX_Cred-310).

Comment (wowkalucky) [reply credentials/settings/base.py:574]: Hardcode vs constants.

File: credentials/apps/verifiable_credentials/models.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        default=settings.SITE_ID,
+        related_name="%(class)ss",
+        on_delete=models.CASCADE,
+    )
+    slug = models.SlugField(
+        max_length=30,
+        default="default",
+        blank=True,
+    )
+    organization = models.ForeignKey(
```
Comment (wowkalucky) [inline credentials/apps/verifiable_credentials/models.py:42]: Did you see [this](https://github.com/raccoongang/credentials/blob/dcc.master/credentials/apps/catalog/models.py#L17)?

**Review by wowkalucky — CHANGES_REQUESTED** [review]

File: credentials/settings/base.py (hunk 3 above)
Comment (Thposbsmt) [reply credentials/settings/base.py]: Removed in this PR due to `OeX_Cred-310`

File: credentials/apps/verifiable_credentials/models.py [hunk 5]
```diff
@@ -1,3 +1,56 @@
...
+
+    .. no_pii:
+    """
+
+    KEY_FIELDS = (
+        "site_id",
+        "slug",
+    )
+    DIGITAL_CREDENTIAL_FORMAT_CHOICES = (
+        (VERIFIABLE_CREDENTIAL_KEY, "Verifiable Credentials"),
```
Comment (wowkalucky) [inline credentials/apps/verifiable_credentials/models.py]: gettext via class attr?

File: credentials/apps/verifiable_credentials/models.py [hunk 6]
```diff
@@ -1,3 +1,56 @@
...
+    organization = models.ForeignKey(
+        Organization,
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+    )
+    digital_credential_format = models.CharField(
+        max_length=128,
+        blank=True,
+        verbose_name="Digital Credential Format",
```
Comment (wowkalucky) [inline credentials/apps/verifiable_credentials/models.py]: gettext

File: requirements/common_constraints.txt [hunk 7]
```diff
@@ -3,6 +3,21 @@
 # See BOM-2721 for more details.
 # Below is the copied and edited version of common_constraints
 
+# This is a temporary solution to override the real common_constraints.txt
```
Comment (wowkalucky) [inline requirements/common_constraints.txt:6]: Seems like a pip-tools bug. Remove duplicates please.

**Review by wowkalucky — APPROVED** [review]
Well done.

**Review by wowkalucky — APPROVED** [review]
Yeahh!

**Pull request merged**